content/docs/refs-and-the-dom.md

```diff
@@ -66,7 +66,7 @@ class CustomTextInput extends React.Component {
 }
 ```
 
-React will call the `ref` callback with the DOM element when the component mounts, and call it with `null` when it unmounts.
+React will call the `ref` callback with the DOM element when the component mounts, and call it with `null` when it unmounts. `ref` callbacks will be called before `componentDidMount`.
```

Comment:
This new wording could be slightly confusing because `ref` callbacks _might_ be called after mount if a new `ref` callback is provided (as is the case with arrow-functions that are recreated each time `render` is run). In that case, React clears the old one and calls the new one _during_ the render phase (before `componentDidMount` / `componentDidUpdate`).

Reply:
What do you think of wording more like:set before mount and update lifecycles"

Reply:
What do you think of wording more like:

> `ref` callbacks are invoked before `componentDidMount` or `componentDidUpdate` lifecycle hooks.

Reply:
Sounds good to me, I'll push an update.